control-plane/agents/src/bin/core/tests/volume/switchover.rs

```diff
@@ -2,22 +2,179 @@
+            &PublishVolume {
+                uuid: vol.uuid().clone(),
+                share: None,
+                target_node: Some(cluster.node(0)),
+                publish_context: HashMap::new(),
+                frontend_nodes: vec![cluster.node(1).to_string()],
+            },
+            None,
+        )
+        .await
+        .expect("failed to publish volume");
+
+    let target_node = vol.state().target.unwrap().node;
+    assert_eq!(target_node, cluster.node(0));
+
+    let vol = vol_client
+        .republish(
+            &RepublishVolume {
+                uuid: vol.uuid().clone(),
+                target_node: Some(cluster.node(1)),
+                share: VolumeShareProtocol::Nvmf,
+                reuse_existing: false,
+                frontend_node: cluster.node(1),
+                reuse_existing_fallback: false,
+            },
+            None,
+        )
+        .await
+        .expect("unable to republish volume");
+
+    let republished_node = vol.state().target.unwrap().node;
+    assert_ne!(
+        republished_node, target_node,
+        "expected nexus node to change post republish"
+    );
+
+    let nexuses = nexus_client
+        .get(Filter::None, None)
+        .await
+        .expect("Could not list nexuses");
+    assert_eq!(nexuses.0.len(), 2, "expected two nexuses after republish");
+
+    cluster
+        .composer()
+        .stop(cluster.node(0).as_str())
+        .await
+        .expect("failed to stop container");
+
+    let nexuses_after_pause = nexus_client
+        .get(Filter::None, None)
+        .await
+        .expect("could not list nexuses");
+    sleep(Duration::from_secs(2)).await;
+
+    assert_eq!(
+        nexuses_after_pause.0.len(),
+        1,
+        "expected 1 nexus after stop"
+    );
+
+    let spec = spec_client
+        .get_specs()
+        .await
+        .expect("expected to retrieve specs");
+
+    let nexus_spec = spec.nexuses;
+    assert_eq!(nexus_spec.len(), 2, "spec should contain 2 nexuses");
+
+    vol_client
+        .destroy(
+            &DestroyVolume {
+                uuid: vol.uuid().clone(),
+            },
+            None,
+        )
+        .await
+        .expect("failed to delete volume");
+
+    let spec = spec_client
+        .get_specs()
+        .await
+        .expect("expected to retrieve specs");
+
+    let nexus_spec = spec.nexuses;
+    assert_eq!(
+        nexus_spec.len(),
+        1,
+        "spec should contain one nexus after vol destroy with old target node unreachable"
+    );
+
+    let n = nexus_spec.get(0).unwrap();
+    assert_eq!(
+        n.status,
+        SpecStatus::Deleting,
+        "failed to mark nexus spec as Deleting"
+    );
+
+    cluster
+        .composer()
+        .start(cluster.node(0).as_str())
+        .await
+        .expect("failed to start container");
+    sleep(Duration::from_secs(5)).await;
+
+    let spec = cluster
+        .rest_v00()
+        .specs_api()
+        .get_specs()
```

Comment:
```suggestion
    let spec = spec_client() .get_specs()
```

Reply:
hmm no, this line is still not changed..

Reply:
oh yes, changed it now :)